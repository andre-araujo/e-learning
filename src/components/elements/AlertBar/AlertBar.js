import React from 'react';
import { node } from 'prop-types';

import {
  Container,
} from './AlertBar.styles';

function AlertBar({
  children,
}) {
  if (!children) return null;

  return (
    <Container>
      {children}
    </Container>
  );
}

AlertBar.defaultProps = {
  children: null,
};

AlertBar.propTypes = {
  children: node,
};

export default AlertBar;
