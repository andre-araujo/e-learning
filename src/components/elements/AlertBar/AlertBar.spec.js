import React from 'react';
import { shallow } from 'enzyme';

import AlertBar from './AlertBar';

describe('AlertBar component', () => {
  it('should match snapshot', () => {
    expect(shallow(<AlertBar />)).toMatchSnapshot();
  });

  it('should render it\'s children', () => {
    const rendered = shallow(
      <AlertBar>
        <p>
          children
        </p>
      </AlertBar>,
    );

    expect(rendered.find('p').text()).toEqual('children');

    expect(rendered).toMatchSnapshot();
  });
});
