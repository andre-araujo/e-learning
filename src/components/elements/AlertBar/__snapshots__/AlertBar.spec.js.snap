// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBar component should match snapshot 1`] = `""`;

exports[`AlertBar component should render it's children 1`] = `
<styled.div>
  <p>
    children
  </p>
</styled.div>
`;
